### DISCUSSION TITLE
how to send message with `Bot` object

### DISCUSSION THREAD

**Discussion by perfection021** [kianahmadian/python-bale-bot#77] [Q&A] [answered]
با سلام . آیا امکان ارسال پیام به گروه به صورت مستقیم هست ؟ 
یعنی دیگه توی آپدیت نباشه و تو یه دستور پیامی که میخوایم رو توی گروه بفرسته

**Comment by kian-ahmadian** [comment]
سلام عزیز بله میتونید
کافی صرفا از طریق کلاس بات اقدام کنید
Bot.send_message()

راهنمای این کد رو میتونید در داکیومنت مشاهده کنید

**Reply by perfection021** [reply]
استفاده کردم ولی اروری میده و متوجهش نمیشم 
RuntimeWarning: coroutine 'Bot.send_message' was never awaited
  Bot.send_message(self=client1,text="s", chat_id="")
RuntimeWarning: Enable tracemalloc to get the object allocation traceback

**Reply by kian-ahmadian** [reply]
میبایست از عبارت await در پشت دستور ارسال پیام استفاده بشه

**Reply by kian-ahmadian** [reply, marked as answer]
@perfection021 
## نمونه کد برای درک بهتر
```python
...
client = bale.Bot(...)
async def _():
     await client.send_message(1234, "hi from python-bale-bot!", ...) # https://docs.python-bale-bot.ir/en/stable/bale.bot.html#bale.Bot.send_message
```

**Reply by perfection021** [reply]
'NoneType' object has no attribute 'request'
Traceback (most recent call last):
  File "C:\Users\ASUS\PycharmProjects\learning project\khabare-vije\bale-bot\venv\lib\site-packages\bale\request\http.py", line 123, in request
    async with self.__session.request(method=method, url=url, **kwargs) as original_response:
AttributeError: 'NoneType' object has no attribute 'request'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\ASUS\PycharmProjects\learning project\khabare-vije\bale-bot\venv\lib\site-packages\pyrogram\dispatcher.py", line 240, in handler_worker
    await handler.callback(self.client, *args)
  File "C:\Users\ASUS\PycharmProjects\learning project\khabare-vije\bale-bot\bale1.py", line 15, in _
    print(await client1.send_message(text="kamal",chat_id=2045627338))
  File "C:\Users\ASUS\PycharmProjects\learning project\khabare-vije\bale-bot\venv\lib\site-packages\bale\bot.py", line 438, in send_message
    response = await self._http.send_message(
  File "C:\Users\ASUS\PycharmProjects\learning project\khabare-vije\bale-bot\venv\lib\site-packages\bale\request\http.py", line 157, in request
    raise HTTPException(error)
bale.error.HTTPException: 'NoneType' object has no attribute 'request'

متن کل ارور اینه اگه ممکنه کمک کنید .

**Reply by kian-ahmadian** [reply]
بنده بخش کوچکی از تکه کد رو براتون نوشتم
دستور ران رو هم میبایست اجرا کنید

**Reply by perfection021** [reply]
import asyncio
from bale import Bot

client1 = Bot(token="")
async def _():
        await client1.send_message(text="kamal",chat_id=2045627338)

asyncio.run(_())
client1.run()

کل کد اینه چیز دیگه ای هم باید اضافه کنم آیا ؟

**Reply by kian-ahmadian** [reply]
**خیر عزیز؛ در صورتی که میخواید از این دستور در ایونت ها استفاده کنید؛ باید صرفا در پایان دستور ران شدن ربات رو استفاده کنید و `asyncio.run(_())` صحیح نیست.**
اما در صورتی که صرفا رباتی رو میخواید که پیامی رو براتون ارسال کنه میتونید از این تکه کد کمک بگیرید:
```python
from bale import Bot
import asyncio
client = Bot("Token")

async def run():
    async with client as bot:
        await bot.send_message(1234, "Hi, python-bale-bot!")

asyncio.run(run())
```

**Reply by perfection021** [reply]
خیلی ممنون کار کرد 🙏🙏
